dunesim/GapChargeTransport/GapChargeTransport_tool.cc

```diff
@@ -0,0 +1,243 @@
+      minY = std::min(minY, bbox.MinY());
+      maxY = std::max(maxY, bbox.MaxY());
+      minZ = std::min(minZ, bbox.MinZ());
+      maxZ = std::max(maxZ, bbox.MaxZ());
+    }
+    std::cout << "The bottom left point (min): (" << minX << ", " << minY << ", " << minZ << ")"
+              << std::endl;
+    std::cout << "The active volume dimensions: x = " << activeVolumeSizeX
+              << ", y = " << activeVolumeSizeY << "z = " << activeVolumeSizeZ << std::endl;
+    std::cout.flush();
+    detector_size_X = maxX - minX;
+    detector_size_Y = maxY - minY;
+    detector_size_Z = maxZ - minZ;
+    nTPCsX = static_cast<int>(detector_size_X / activeVolumeSizeX + 0.5);
+    nTPCsY = static_cast<int>(detector_size_Y / activeVolumeSizeY + 0.5);
+    nTPCsZ = static_cast<int>(detector_size_Z / activeVolumeSizeZ + 0.5);
+    average_gap_Z = (detector_size_Z - (double)nTPCsZ * activeVolumeSizeZ) / (double)(nTPCsZ - 1);
+    average_gap_Y = (detector_size_Y - (double)nTPCsZ * activeVolumeSizeY) / (double)(nTPCsY - 1);
+    average_gap_X = (detector_size_X - (double)nTPCsX * activeVolumeSizeX) / (double)(nTPCsX - 1);
+    if (average_gap_Z > 2 || average_gap_Y > 2 || average_gap_X > 2) {
+      std::cout << "WARNING: The gaps are weirdly huge: y = " << average_gap_Y
+                << ", z = " << average_gap_Z << ", x = " << average_gap_X << std::endl;
+      std::cout.flush();
+    }
+  }
+
+  //here the parameters of the gap are calculated
+  GapInfo GapChargeTransport::Gap(double x, double y, double z) const
+  {
+    int id_z = static_cast<int>((z - minZ) / (activeVolumeSizeZ));
+    int id_y = static_cast<int>((y - minY) / (activeVolumeSizeY));
+    int id_x = static_cast<int>((x - minX) / (activeVolumeSizeX));
+    std::cout << "The bottom left point (min): (" << minX << ", " << minY << ", " << minZ << ")"
+              << std::endl;
+    std::cout << "The active volume dimensions: x = " << activeVolumeSizeX
+              << ", y = " << activeVolumeSizeY << ", z = " << activeVolumeSizeZ << std::endl;
+    std::cout << "id_x = " << id_x << ", id_y = " << id_y << ", id_z = " << id_z << std::endl;
+    std::cout.flush();
+    int id_total = (id_x) * (nTPCsY) + (id_y) + (id_z) * (nTPCsX * nTPCsY);
+    GapInfo::GapType type = GapInfo::GapType::NotAGap;
+    GapInfo::MovingDirection direction = GapInfo::MovingDirection::LEFT;
+    double left = 0;
+    double right = 0;
+    if (z < minZ || z > maxZ || y < minY || y > maxY || x < minX || x > maxX ||
+        id_total >= totalTPCs) {
+      type = GapInfo::GapType::NotInDetectorVolume;
+    }
+    else {
+      const geo::TPCGeo& tpc = geom->TPC(geo::TPCID(0, id_total));
+      geo::BoxBoundedGeo bbox = tpc.ActiveBoundingBox();
+      std::cout << "id_total = " << id_total << ", MaxX = " << bbox.MaxX()
+                << ", MaxY = " << bbox.MaxY() << ", MaxZ = " << bbox.MaxZ() << std::endl;
+      std::cout.flush();
+
+      if (z < bbox.MinZ()) { //not in the "right" active volume
```

Comment:
I'm struggling to parse when this fires, as it looks like the associated TPC ID is always upstream of the point being assessed (from line 68)